drivers/gles3/shaders/scene.glsl

```diff
@@ -364,10 +367,11 @@ void light_compute(vec3 N, vec3 L, vec3 V, vec3 light_color, bool is_directional
 	float specular_brdf_NL = 0.0;
 	// Normalized blinn always unless disabled.
 	vec3 H = normalize(V + L);
-	float cNdotH = clamp(dot(N, H), 0.0, 1.0);
+	float cNdotH = clamp(A + dot(N, H), 0.0, 1.0);
+	float cLdotH = clamp(A + dot(L, H), 0.0, 1.0);
 	float shininess = roughness_to_shininess(roughness);
 	float blinn = pow(cNdotH, shininess);
-	blinn *= (shininess + 2.0) * (1.0 / (8.0 * M_PI)) * cNdotL;
+	blinn *= (shininess + 8.0) * (1.0 / (8.0 * M_PI)) * (0.25 / max(0.0001, cLdotH * cLdotH)) * cNdotL * specular_amount;
```

Comment:
Where does this extra term come from? I don't think it is necessary for a simple blinn NDF. 

IIRC the current normalization factor comes from http://www.thetenthplanet.de/archives/255 and it is energy conserving. I wouldn't change it without a good reason

Reply:
This is the Geometry term mentioned in the fourth point, and is the same method we use for clearcoat specular. It was added to more closely match per-pixel shading with minimal extra cost, but we can drop this if that's not a priority.

Reply:
Oh, I see. I would drop it unless it makes a huge difference. Seeing as we already don't use the Fresnel term, I don't think the extra cost is worth it just to be slightly more correct. Fresnel makes way more of a difference, but you don't really miss it with vertex shading.

Reply:
It is relatively subtle so yeah, I'll go ahead and drop this.